product_configurator/models/product.py

```diff
@@ -440,6 +446,44 @@ def values_available(self, attr_val_ids, sel_val_ids):
         return avail_val_ids
 
     @api.multi
+    def find_default_value(self, selectable_value_ids, value_ids):
+        """Based on the current values, which of the available template value ids
+            is the best default value to use.
+
+            :param selectable_value_ids: list of product.attribute.value
+                object already trimmed down as selectable, for one
+                attribute line.
+            :param value_ids: list of attribute value ids already chosen
+
+            :returns: The first matched default id
+
+        """
+        self.ensure_one()
+
+        if not selectable_value_ids:
+            return False
+        # assume all values are from the same attribute line - they should be!
+        default_lines = self.config_default_ids.filtered(
+            lambda l: set(l.value_ids.ids) & set(selectable_value_ids)
+        )
+
+        for default_line in default_lines:
+            if not default_line.domain_id:
+                # No domain - always considered true. Use this.
+                break
+            domains = default_line.mapped('domain_id').compute_domain()
+            if self.validate_domains_against_sels(domains, value_ids):
+                # Domain OK, use this
+                break
+        else:
+            # parsed all lines without a match
+            return False
+        # pick one at random...
+        return (
+            set(default_line.value_ids.ids) & set(selectable_value_ids)
+        ).pop()
```

Comment:
This will not work for multi, the client expects a list. For instance, adding options:armrest will cause a js error in AbstractManyField.set_value()